Add various REST API docs (emails, files, marketing messages, notes, pages, roles) (#139)
mautic/developer-documentation-new: docs/rest_api/users.rst

.. vale on

``POST /users/new``

**POST Parameters**

.. list-table::
   :header-rows: 1

   * - Name
     - Type
     - Description
   * - ``username``
     - string
     - Username that's used to log into Mautic.
   * - ``firstName``
     - string
     - First Name of the User
   * - ``lastName``
     - string
     - Last Name of the User
   * - ``email``
     - string
     - Email of the User
   * - ``position``
     - string
     - User's position title
   * - ``role``
     - int
     - Role ID
   * - ``timezone``
     - string
     - Timezone of the User
   * - ``onlineStatus``
     - string
     - Online status of the User
   * - ``signature``
     - string
     - Signature of the User for use in Emails
   * - ``plainPassword``
     - array
     - array of plain password as in the example

**Response**

``Expected Response Code: 201``

**Properties**

Same as `Get User <#get-user>`_.

.. vale off

Edit User
*********

.. vale on

.. code-block:: php

   <?php

   $id   = 1;
   $data = array(
       'lastName' => 'Doeboe',
   );

   // Create new a User of ID 1 isn't found?
   $createIfNotFound = true;

   $user = $userApi->edit($id, $data, $createIfNotFound);

> Edit a new User. User that this supports PUT or PATCH depending on the desired behavior.

Reviewer: ```suggestion
Edit a new User. Note that this supports PUT or PATCH depending on the desired behavior.
```

If ``PATCH``, the expected response code is ``200``.

**Properties**

Same as `Get User <#get-user>`_.

.. vale off

Delete User
************

.. vale on

.. code-block:: php

   <?php

   $user = $userApi->delete($id);

Delete a User.

.. vale off

**HTTP Request**

.. vale on

``DELETE /users/ID/delete``

**Response**

``Expected Response Code: 200``

**Properties**

Same as `Get User <#get-user>`_.

.. vale off

Get Self User
*************

.. vale on

.. code-block:: php

   <?php

   $user = $userApi->getSelf();

Get a self User.

.. vale off

**HTTP Request**

.. vale on

``GET /users/self``

**Response**

``Expected Response Code: 200``

**Properties**

Same as `Get User <#get-user>`_.

.. vale off

Check User Permissions
**********************

.. vale on

.. code-block:: php

   <?php
   $permission = array('user:users:create', 'user:users:edit');
   $user = $userApi->checkPermission($id, $permission);

Get a self User.

.. vale off

**HTTP Request**

.. vale on

``GET /users/ID/permissioncheck``

**Response**

``Expected Response Code: 200``

.. code-block:: json

   {
     "user:users:create":true,
     "user:users:edit":true
   }

**Properties**

> array of requested permissions of string in case of just one

Reviewer: ```suggestion
An array of requested permissions, or a string where returning only one value.
```

I think this is what you mean but please check to be sure of the intended context!

Author: I'll test this endpoint in a sec and will update the description here. Thanks for noticing!
